pkgs/build-support/buildenv/builder.pl

```diff
@@ -117,10 +117,22 @@ sub addPkg {
     }
 }
 
+# Read packages list.
+my $pkgs;
+
+if (exists $ENV{"pkgsPath"}) {
+  local $/ = undef;
+  open FILE, $ENV{"pkgsPath"};
+  binmode FILE;
+  $pkgs = <FILE>;
+  close FILE;
+} else {
+  $pkgs = $ENV{"pkgs"}
+}
```

Comment:
It indeed works (if I understand correctly, that's because `builtins.toJSON` returns one long line), thanks! Updated.